test/e2e-go/restAPI/restClient_test.go

```diff
@@ -2172,141 +2189,1102 @@ func TestMaxDepthAppWithPCTrace(t *testing.T) {
-		{Pc: 76},
-		{Pc: 78},
-		{Pc: 79},
-		{Pc: 81},
-		{Pc: 83},
-		{Pc: 85},
-		{Pc: 87},
-		{Pc: 89, SpawnedInners: &[]uint64{0}},
-		{Pc: 90},
-		{Pc: 91},
-		{Pc: 92},
-		{Pc: 94},
-		{Pc: 95},
-		{Pc: 97},
-		{Pc: 99},
-		{Pc: 103},
-		{Pc: 104},
-		{Pc: 106},
-		{Pc: 113},
-		{Pc: 116},
-		{Pc: 117},
-		{Pc: 118},
-		{Pc: 119},
-		{Pc: 121},
-		{Pc: 122},
-		{Pc: 123},
-		{Pc: 125},
-		{Pc: 128},
-		{Pc: 129},
-		{Pc: 130},
-		{Pc: 131},
-		{Pc: 132},
-		{Pc: 134},
-		{Pc: 136},
-		{Pc: 138},
-		{Pc: 139},
-		{Pc: 141},
-		{Pc: 143},
-		{Pc: 145, SpawnedInners: &[]uint64{1, 2}},
-		{Pc: 146},
-		{Pc: 72},
-		{Pc: 73},
-	}
-
-	finalDepthTrace := []model.SimulationOpcodeTraceUnit{
-		{Pc: 1},
-		{Pc: 6},
-		{Pc: 8},
-		{Pc: 9},
-		{Pc: 10},
-		{Pc: 13},
-		{Pc: 15},
-		{Pc: 16},
-		{Pc: 17},
-		{Pc: 21},
-		{Pc: 23},
-		{Pc: 25},
-		{Pc: 27},
-		{Pc: 29},
-		{Pc: 31},
-		{Pc: 33},
-		{Pc: 35},
-		{Pc: 37},
-		{Pc: 39},
-		{Pc: 41},
-		{Pc: 43},
-		{Pc: 45},
-		{Pc: 47},
-		{Pc: 48},
-		{Pc: 50},
-		{Pc: 51},
-		{Pc: 53},
-		{Pc: 54},
-		{Pc: 56},
-		{Pc: 59},
-		{Pc: 60},
-		{Pc: 61},
-		{Pc: 62},
-		{Pc: 63},
-		{Pc: 66},
-		{Pc: 67},
-		{Pc: 68},
-		{Pc: 69},
-		{Pc: 72},
-		{Pc: 73},
+		{
+			Pc: 1,
+		},
+		// txn ApplicationID
+		{
+			Pc: 6,
+			Additions: &[]model.StackValue{
+				{
+					Type: uint64(basics.TealUintType),
+				},
+			},
+		},
+		// int 0
+		{
+			Pc: 8,
+			Additions: &[]model.StackValue{
+				{
+					Type: uint64(basics.TealUintType),
+				},
+			},
+		},
+		// ==
+		{
+			Pc:        9,
+			Deletions: toPtr[uint64](2),
+			Additions: &[]model.StackValue{
+				{
+					Type: uint64(basics.TealUintType),
+					Uint: toPtr[uint64](1),
+				},
+			},
+		},
+		// bnz main_l6
+		{
+			Pc:        10,
+			Deletions: toPtr[uint64](1),
+		},
+		// int 1
+		{
+			Pc: 149,
+			Additions: &[]model.StackValue{
+				{
+					Type: uint64(basics.TealUintType),
+					Uint: toPtr[uint64](1),
+				},
+			},
+		},
+		// return
+		{
+			Pc: 150,
+			Additions: &[]model.StackValue{
+				{
+					Type: uint64(basics.TealUintType),
+					Uint: toPtr[uint64](1),
+				},
+			},
+			Deletions: toPtr[uint64](1),
+		},
+	}
+
+	boolToUint64 := func(b bool) uint64 {
+		if b {
+			return 1
+		}
+		return 0
+	}
+
+	const NumArgs = 1
+
+	recursiveLongOpcodeTrace := func(appID basics.AppIndex, layer int) *[]model.SimulationOpcodeTraceUnit {
+		return &[]model.SimulationOpcodeTraceUnit{
+			{
+				Pc: 1,
+			},
+			// txn ApplicationID
+			{
+				Pc: 6,
+				Additions: &[]model.StackValue{
+					{
+						Type: uint64(basics.TealUintType),
+						Uint: valToNil(toPtr(uint64(appID))),
+					},
+				},
+			},
+			// int 0
+			{
+				Pc: 8,
+				Additions: &[]model.StackValue{
+					{
+						Type: uint64(basics.TealUintType),
+					},
+				},
+			},
+			// ==
+			{
+				Pc:        9,
+				Deletions: toPtr[uint64](2),
+				Additions: &[]model.StackValue{
+					{
+						Type: uint64(basics.TealUintType),
+						Uint: valToNil(toPtr(boolToUint64(appID == 0))),
```

Comment:
nit: at this point we know the appID cannot be 0, so you could just pass in false?

Reply:
right, see [per pr response](https://github.com/algorand/go-algorand/pull/5440/commits/e02b34c91c659836b3aec983577be3bb898cf09a)